libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManagerInternal.swift

```diff
@@ -558,4 +1000,56 @@ public class NativeLoginManagerInternal: NSObject, NativeLoginManager {
             return .invalidCredentials
         }
     }
+    
+    // MARK: Private String Parameter Validation
+    
+    /// Validates a string is a valid email address.
+    /// - Parameters:
+    ///   - string: The string to validate
+    /// - Returns: nil for valid email addresses - The invalid email login result otherwise
+    ///
+    func mapInvalidEmailAddressToResult(_ string: String) -> NativeLoginResult? {
+        let detector = try? NSDataDetector(
+            types: NSTextCheckingResult.CheckingType.link.rawValue
+        )
+        let range = NSRange(
+            string.startIndex..<string.endIndex,
+            in: string
+        )
+        let matches = detector?.matches(
+            in: string,
+            options: [],
+            range: range
+        )
+        guard let match = matches?.first, matches?.count == 1 else {
+            return .invalidEmail
+        }
+        guard match.url?.scheme == "mailto", match.range == range else {
+            return .invalidEmail
+        }
+        
+        return nil
+    }
+    
+    /// Validates a string is a valid password.
+    /// - Parameters:
+    ///    - string: The string to validate
+    /// - Returns: nil for valid passwords - The invalid password login result otherwise
+    func mapInvalidPasswordToResult(_ string: String) -> NativeLoginResult? {
+        switch (!isValidPassword(password: string.trimmingCharacters(in: .whitespacesAndNewlines))) {
```

Comment:
potentially more of a style thing, why switch on a boolean? are other cases coming in?

Reply:
I'm so used to everything being a `when` in Kotlin instead of `if` that this just sort of happened on muscle memory.  It probably _could_ be an `if` instead.  Shall we "switch" it to "if"? (get it? 🤣)

Reply:
sounds good and of course props for the pun :P